### PR TITLE
[Feature] Manual Sign up

### PR DESCRIPTION
**Pull request by f-morgado**
Relates to #109 

## Screenshots (if visual changes)
image
image
image

## Proposed Changes

  - Verification if an email is already registered and if so, send the user back to log in page
  - New logo and banner on the main frame
  - verification on all the fields necessary for the sign up
  - SSO and non SSO mail verification

### CODE REVIEW

**Review by nunocaseiro — COMMENTED** [review]
The backend package-lock.json should be commited as we only use the one that is in the project root folder

File: frontend/components/Primitives/Input.tsx [hunk 1]
```diff
@@ -243,6 +243,7 @@ const Input: React.FC<InputProps> = ({
   const handleOnClickIcon = () => {
     if (type === "text") return;
     setType(currentType === "password" ? "text" : "password");
+    console.log(icon);
```
Comment (nunocaseiro) [inline frontend/components/Primitives/Input.tsx:246]: You should clean all the console.log

File: frontend/components/auth/SignUp/SignUpForm.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        console.log(" entrou onsucess", data);
+        if (data === "az") {
+          return setShowSignUp(SignUpEnum.SIGN_UP_OPTIONS);
+        }
+        if (data === false) {
+          setShowSignUp(SignUpEnum.REGISTER);
+        }
+        setValueHelperText(" This email already exists");
+        setValueState(true);
+      },
```
Comment (nunocaseiro) [inline frontend/components/auth/SignUp/SignUpForm.tsx:53]:
```suggestion
      onSuccess: (data) => {
        console.log(" entrou onsucess", data);
        if (data === "az") {
         setShowSignUp(SignUpEnum.SIGN_UP_OPTIONS);
         return;
        }
        if (data === false) {
          setShowSignUp(SignUpEnum.REGISTER);
          return;
        }
        setValueHelperText(" This email already exists");
        setValueState(true);
      },
```

File: frontend/components/auth/SignUp/SignUpForm.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    setEmailName({ goback: false, email });
+  };
+
+  return (
+    <FormProvider {...methods}>
+      <StyledForm
+        direction="column"
+        style={{ width: "100%" }}
+        onSubmit={methods.handleSubmit(({ email }) => {
+          console.log(" on submit", email);
```
Comment (nunocaseiro) [inline frontend/components/auth/SignUp/SignUpForm.tsx:85]: Clear the console.log

File: frontend/components/Primitives/Checkbox.tsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ const StyledCheckbox = styled(CheckboxPrimitive.Root, {
         borderColor: "$primaryBase",
         backgroundColor: "transparent",
         '&[data-state="checked"]': {
-          backgroundColor: "$primaryBase",
+          //  backgroundColor: "$primaryBase",
```
Comment (nunocaseiro) [inline frontend/components/Primitives/Checkbox.tsx:24]: remove the comment please

File: frontend/components/Primitives/Checkbox.tsx [hunk 5]
```diff
@@ -52,42 +52,40 @@ const StyledCheckbox = styled(CheckboxPrimitive.Root, {
...
-
 const Checkbox: React.FC<{
   label: string;
   id: string;
   variant?: "default" | "error";
   checked?: boolean | "indeterminate";
   disabled?: boolean;
   size: "12" | "14" | "16";
+  setCheckedTerms: Dispatch<React.SetStateAction<boolean>>;
+  // eslint-disable-next-line react/require-default-props
```
Comment (nunocaseiro) [inline frontend/components/Primitives/Checkbox.tsx:72]: remove this eslint disable rule

File: frontend/components/Primitives/Checkbox.tsx [hunk 6]
```diff
@@ -52,42 +52,40 @@ const StyledCheckbox = styled(CheckboxPrimitive.Root, {
...
     disabled: false,
-    handleChange: () => {},
   };
 
   const [currentCheckValue, setCurrentCheckValue] = useState<boolean | undefined | "indeterminate">(
     checked
   );
-
   const handleCheckedChange = (isChecked: boolean | "indeterminate") => {
+    console.log("click ;)");
```
Comment (nunocaseiro) [inline frontend/components/Primitives/Checkbox.tsx:85]: console.log eheh

File: frontend/components/auth/SignUp/RegisterForm.tsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+const RegisterForm: React.FC<RegisterFormProps> = ({
+  setShowSignUp,
+  emailName,
+  setCurrentTab,
+  setEmailName,
+}) => {
+  const setToastState = useSetRecoilState(toastState);
+  const [checkedTerms, setCheckedTerms] = useState(false);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (nunocaseiro) [inline frontend/components/auth/SignUp/RegisterForm.tsx:48]: remove this eslint disable

File: frontend/components/auth/SignUp/SignUpForm.tsx (hunk 2 above)
Comment (nunocaseiro) [reply frontend/components/auth/SignUp/SignUpForm.tsx:53]: and remove the eslint

File: frontend/components/auth/SignUp/SignUpForm.tsx [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+        }
+        if (data === false) {
+          setShowSignUp(SignUpEnum.REGISTER);
+        }
+        setValueHelperText(" This email already exists");
+        setValueState(true);
+      },
+
+      onError: () => {
+        console.log("entrou onerror");
```
Comment (nunocaseiro) [inline frontend/components/auth/SignUp/SignUpForm.tsx:56]: clear console.log

File: frontend/components/auth/SignUp/SignUpForm.tsx [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+    mode: "onSubmit",
+    reValidateMode: "onSubmit",
+    defaultValues: {
+      email: emailName.email,
+    },
+    resolver: zodResolver(SchemaEmail),
+  });
+
+  const handleCheckUserExists = async (email: string) => {
+    console.log("handleCheckUserExists");
```
Comment (nunocaseiro) [inline frontend/components/auth/SignUp/SignUpForm.tsx:75]: clear

File: frontend/components/auth/SignUp/SignUpOptionsForm.tsx [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+  emailName: string;
+  setEmailName: Dispatch<React.SetStateAction<{ email: string; goback: boolean }>>;
+}
+
+const SignUpOptionsForm: React.FC<SignUpOptionsFormProps> = ({
+  setShowSignUp,
+  emailName,
+  setEmailName,
+}) => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (nunocaseiro) [inline frontend/components/auth/SignUp/SignUpOptionsForm.tsx:23]: remove this line

File: frontend/components/auth/SignUp/SignUpOptionsForm.tsx [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+  const [loginErrorCode, setLoginErrorCode] = useState(-1);
+  const { loginAzure } = useUser(setLoginErrorCode);
+
+  const handleEmailChange = () => {
+    setEmailName((prev) => ({ ...prev, goback: true }));
+    return setShowSignUp(SignUpEnum.SIGN_UP);
+  };
+
+  // const [ShowLogIn, setShowLogIn] = useState(false);
```
Comment (nunocaseiro) [inline frontend/components/auth/SignUp/SignUpOptionsForm.tsx:32]: and this

File: frontend/components/auth/SignUp/SignUpOptionsForm.tsx [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+  return (
+    <Container direction="column">
+      <Text size="md">
+        The email
+        <Text size="md" fontWeight="medium">
+          {` ${emailName} `}
+        </Text>
+        {/* {`${textContent}`} */}
```
Comment (nunocaseiro) [inline frontend/components/auth/SignUp/SignUpOptionsForm.tsx:40]: and this

**Review by nunocaseiro — CHANGES_REQUESTED** [review]

File: frontend/components/auth/SignUp/SignUpForm.tsx (hunk 2 above)
Comment (f-morgado) [reply frontend/components/auth/SignUp/SignUpForm.tsx]: fixed

File: frontend/components/Primitives/Checkbox.tsx (hunk 4 above)
Comment (f-morgado) [reply frontend/components/Primitives/Checkbox.tsx]: fixed

File: frontend/components/auth/SignUp/SignUpOptionsForm.tsx (hunk 12 above)
Comment (f-morgado) [reply frontend/components/auth/SignUp/SignUpOptionsForm.tsx]: done

File: frontend/components/auth/SignUp/SignUpOptionsForm.tsx (hunk 11 above)
Comment (f-morgado) [reply frontend/components/auth/SignUp/SignUpOptionsForm.tsx]: done

File: frontend/components/auth/SignUp/SignUpForm.tsx (hunk 8 above)
Comment (f-morgado) [reply frontend/components/auth/SignUp/SignUpForm.tsx]: done

**Comment by nunocaseiro** [comment]
@all-contributors please add @f-morgado  for code and doc

**Review by nunocaseiro — APPROVED** [review]
LGTM 🚀

**Comment by f-morgado** [comment]
ty for all the time you took helping me Nuno 🦾

**Pull request merged**